ros_buildfarm/doc_job.py

```diff
@@ -407,34 +408,49 @@ def configure_doc_independent_job(
 
 def _get_doc_independent_job_config(
         config, config_url, doc_build_name, build_file):
-    template_name = 'doc/doc_independent_job.xml.em'
-
-    repository_args, script_generating_key_files = \
-        get_repositories_and_script_generating_key_files(config=config)
-
     job_data = {
         'job_priority': build_file.jenkins_job_priority,
         'node_label': get_node_label(build_file.jenkins_job_label),
 
         'ros_buildfarm_repository': get_repository(),
 
-        'doc_repositories': build_file.doc_repositories,
-
-        'script_generating_key_files': script_generating_key_files,
-
         'config_url': config_url,
         'doc_build_name': doc_build_name,
-        'repository_args': repository_args,
-
-        'upload_user': build_file.upload_user,
-        'upload_host': build_file.upload_host,
-        'upload_root': build_file.upload_root,
 
         'notify_emails': build_file.notify_emails,
 
         'timeout_minutes': build_file.jenkins_job_timeout,
-
-        'credential_id': build_file.upload_credential_id,
     }
+
+    if build_file.documentation_type == 'make_target':
```

Comment:
I feel somewhat hesitant to tie the upload mechanism to the documentation type. It's possible that an external_site would still publish via ssh/rsync or that we may want to publish make_target docs via git.

This will need further discussion because I think ideally we'd preserve support for the existing upload configuration while deprecating it in favor of a nested structure that we can validate.

Something like below for the git publisher
```yaml
upload:
  type: git
  repository_url: git@github.com:ros-infrastructure/doc-target.git
  repository_branch: gh-pages
  credential_id: <credential_id>
```

and for the doc_independent job that uses make_target
```yaml
upload:
  type: rsync_ssh
  host: some-docs-server.ros.org
  root: /var/www/docs
  user: docsbot
  credential_id: <credential_id>
```

Reply:
Makes perfect sense and I do like it better. However, if we go down that route, no repository management can take place inside the container. Not even a git commit.

In any case, I fear that improvement may delay this new functionality reaching production. What do you think?

Reply:
As much as I want to take the time to improve things while we're touching them, I think the change is either a pre-requisite to this PR or just an enhancement to follow up on and not something that should be included in this work.